i6_models/parts/frontend/vgg_act.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import annotations
+
+__all__ = [
+    "VGG4LayerActFrontendV1",
+    "VGG4LayerActFrontendV1Config",
+]
+
+from dataclasses import dataclass
+from typing import Callable, Optional, Tuple, Union
+
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+from .common import IntTupleIntType, _get_padding, _mask_pool, _get_int_tuple_int
+
+
+@dataclass
+class VGG4LayerActFrontendV1Config(ModelConfiguration):
+    """
+    Attributes:
+        conv1_channels: number of channels for first conv layer
+        conv2_channels: number of channels for second conv layer
+        conv3_channels: number of channels for third conv layer
+        conv4_channels: number of channels for fourth conv layer
+        conv_kernel_size: kernel size of conv layers
+        conv_padding: padding for the convolution
+        pool1_kernel_size: kernel size of first pooling layer
+        pool1_stride: stride of first pooling layer
+        pool1_padding: padding for first pooling layer
+        pool2_kernel_size: kernel size of second pooling layer
+        pool2_stride: stride of second pooling layer
+        pool2_padding: padding for second pooling layer
+        activation: activation function at the end
+        linear_input_dim: input size of the final linear layer
+        linear_output_dim: output size of the final linear layer
+    """
+
+    conv1_channels: int
+    conv2_channels: int
+    conv3_channels: int
+    conv4_channels: int
+    conv_kernel_size: IntTupleIntType
+    conv_padding: Optional[IntTupleIntType]
+    pool1_kernel_size: IntTupleIntType
+    pool1_stride: Optional[IntTupleIntType]
+    pool1_padding: Optional[IntTupleIntType]
+    pool2_kernel_size: IntTupleIntType
+    pool2_stride: Optional[IntTupleIntType]
+    pool2_padding: Optional[IntTupleIntType]
+    activation: Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]
+    linear_input_dim: Optional[int]
```

Comment:
Why is `linear_input_dim` configurable? This is sth which is implied by the other settings, via the input dimension (F), or not? I think I rather would configure the `in_features` (F) for this module, instead of this.

Reply:
I think these are two points.. 
why would you make the `in_features` configurable.. for our use case this should normally not be necessary, right?
`linear_input_dim` is for the implicit user check.. necessary to specify what is expected, if there is a mismatch that points to a bug.

Reply:
I prefer keeping this implicit check.

Reply:
So I think the opinion of others would be good here.

I think this way is inconsistent to all other PyTorch modules, where you specify the input and output feature dimension (classical example is `nn.Linear`).

Reply:
The module has the input and output feature dimension. This would be additionally.

Reply:
But output dimensions are usually called `out_features`, but here called `linear_output_dim`.
And no, you do not have any option for the input dimension here?

Reply:
I have `in_features`, `linear_input_dim, and `out_features`, so I am confused what you mean?

Reply:
Ah, sorry, I was not looking at the latest version. Yes, you have that.

So then the remaining thing is just `linear_input_dim`. I think it would be good to get the opinions of some others. Whether people actually realize this is nothing to configure but just to perform a sanity check.